Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -145,8 +145,217 @@ public struct HTTPResponseHead: Equatable {
+    ///
+    /// - parameters:
+    ///     - name: The name of the header.
+    /// - returns: The indicies of the values for the given name.
+    func findAll(name: String) -> [HTTPHeaderIndex] {
+        guard !self.indicies.isEmpty else {
+            return []
+        }
+
+        let utf8 = name.utf8
+        let stringLength = utf8.count
+        var idx = 0
+        var array: [HTTPHeaderIndex] = []
+        while idx < self.indicies.count {
+            let nameIdx = self.indicies[idx]
+            if stringLength == nameIdx.length && self.buffer.equalCaseInsensitiveASCII(seq: utf8, at: nameIdx) {
+                array.append(self.indicies[idx + 1])
+            }
+            idx += 2
+        }
+        return array
+    }
+
+    /// Add a header.
+    ///
+    /// - parameters:
+    ///     - name: The name for the header.
+    ///     - value: The value for the header.
+    func add(name: String, value: String) {
+        let nameStart = self.buffer.writerIndex
+        let nameLength = self.buffer.write(string: name)!
+        self.buffer.write(staticString: headerSeparator)
+        let valueStart = self.buffer.writerIndex
+        let valueLength = self.buffer.write(string: value)!
+        self.indicies.append(HTTPHeaderIndex(start: nameStart, length: nameLength))
+        self.indicies.append(HTTPHeaderIndex(start: valueStart, length: valueLength))
+        self.buffer.write(staticString: crlf)
+        assert(indicies.count % 2 == 0)
+    }
+
+    /// Remove all headers for the given name.
+    ///
+    /// - parameters:
+    ///     - name: The header name.
+    func remove(name: String) -> Bool {
+        guard !self.indicies.isEmpty else {
+            return false
+        }
+
+        let utf8 = name.utf8
+        let stringLength = utf8.count
+        var idx = 0
+        var array: [Int] = []
+        while idx < self.indicies.count {
+            let nameIdx = self.indicies[idx]
+            if stringLength == nameIdx.length && self.buffer.equalCaseInsensitiveASCII(seq: utf8, at: nameIdx) {
+                array.append(idx)
+            }
+            idx += 2
+        }
+
+        guard !array.isEmpty else {
+            return false
+        }
+        array.reversed().forEach {
+            self.indicies.removeSubrange($0...$0 + 1)
+        }
+        assert(indicies.count % 2 == 0)
+        return true
+    }
 
+    var description: String {
+        var i = 0
+        var str = ""
+        for idx in self.indicies {
+            str.append(string(idx: idx))
+            if i % 2 == 0 {
+                str.append(": ")
+            } else {
+                str.append("\r\n")
+            }
+            i += 1
+        }
+        return str
+    }
+
+    /// Write content of this storage to buffer.
+    ///
+    /// - parameters:
+    ///     - into: The `ByteBuffer` into which we should write the content.
+    func write(into: inout ByteBuffer, continous: Bool) {
```

Comment:
How are you supposed to know if this is continuous? (Also, s/continous/continuous/)